mcs/class/corlib/System.Reflection/AssemblyName.cs

```diff
@@ -151,11 +151,7 @@ internal AssemblyName (SerializationInfo si, StreamingContext sc)
 				if (codebase == null)
 					return null;
 
-#if NETCORE
-				throw new NotImplementedException ();
-#else
 				return Mono.Security.Uri.EscapeString (codebase, false, true, true);
```

Comment:
I am not a big fan of bringing back any dependency to Mono.[In]Security. CoreCLR has the implementation in managed code, so it should probably be moved to shared CoreLib partition and reused:

https://github.com/dotnet/coreclr/blob/dcf9f36c21022b8fce374559ec5fea8d11a0852b/src/System.Private.CoreLib/src/System/Reflection/AssemblyName.cs#L396-L577

(CoreRT throws PlatformNotSupportedException)

Reply:
Its just one file, we can move it to corelib for now.

Reply:
I guess it should not block this PR, but I wanted to bring it to attention. I can do the code shuffling between repositories later.

Reply:
Yes, AssemblyName should be in shared/